### ISSUE TITLE
[RG] Debugging report generator errors related to horizontal aggregation when columns have different lengths

### ISSUE THREAD

**Issue by ltp23** [RuminantFarmSystems/RuFaS#1498]
I'd like to collect examples of errors caused by the report generator. It would be very helpful if you could provide the following:
- Filters that are not working
- Current branch and scenario
- Error messages

**Comment by jadamchick** [comment]
Thanks Loi! 
Attaching here: 
- the filter I am using for pilot testing
[report_pilots 3Mar.json](https://github.com/RuminantFarmSystems/MASM/files/14857595/report_pilots.3Mar.json)

- three errors from running the same farm (two report-related, the other "list index out of range")
[c2f7_3of4_final_02Apr_errors_02-Apr-2024_Tue_09-20-40.json](https://github.com/RuminantFarmSystems/MASM/files/14857588/c2f7_3of4_final_02Apr_errors_02-Apr-2024_Tue_09-20-40.json)
[c2f7_6of4_final_02Apr_errors_02-Apr-2024_Tue_09-33-57.json](https://github.com/RuminantFarmSystems/MASM/files/14857589/c2f7_6of4_final_02Apr_errors_02-Apr-2024_Tue_09-33-57.json)
[c2f7_8of4_final_02Apr_errors_02-Apr-2024_Tue_18-20-14.json](https://github.com/RuminantFarmSystems/MASM/files/14857592/c2f7_8of4_final_02Apr_errors_02-Apr-2024_Tue_18-20-14.json)




Branch: pilot_testing_Dec2023
Metadata: c2f7_metadata.json

I ran this farm 8 times on this day (attempting to get 4 good runs). The results were: 
- run 1: no errors
- runs 2 & 3: report generation errors (they were very similar, though I can't swear identical). Error file from run 3 attached
- runs 4 & 5: no errors
- run 6: report generation error, some slightly different than previously (eg which group of animals it couldn't run enteric methane for). Error file attached. 
- run 7: no errors
- run 8: a completely different error, not report generation (likely not related to this PR, but attaching it here in case it is useful)

**Comment by jadamchick** [comment]
Hi Loi - one more update. 
C2f12 (on pilot_testing_Dec2023) was also repeatedly throwing this type of error (one file uploaded below) - using the same filter as above. I noticed that it was creating many new pens, so I updated the stocking density + stall numbers as follows: 
- stocking density from all pens: from 2.5 --> 2 (since the new "max stocking density" was implemented, this was now getting changed to the default 1.2, causing a need for many more new pens!)
- number of stalls in pen 0 (calf): from 80 -> 120
- number of stalls in pen 1 (growing): 800 -> 1200 

It then ran fine for me four times consecutively. (which was all I needed). 
Though, worth noting: it did still create a bunch of new pens in some but not all of those simulations. 

Maybe this is just a red herring, or maybe it is related to the new pen generation? 
[c2f12_5of4_final_03Apr_errors_03-Apr-2024_Wed_16-53-59.json](https://github.com/RuminantFarmSystems/MASM/files/14871594/c2f12_5of4_final_03Apr_errors_03-Apr-2024_Wed_16-53-59.json)
